### ISSUE TITLE
[cmake] cmake does not enable testing

### ISSUE THREAD

**Issue by cdluminate** [asmjit/asmjit#284]
By convention when a user finished `cmake <...>; make <...>`, they can subsequently run `make test` or `ninja test` to run the unit tests. The current cmakefile does not enable that. Following is a reference fix:

```patch
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 81c928c..0135e94 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -419,6 +419,7 @@ if (NOT ASMJIT_EMBED)
 
   # Add 'asmjit' tests.
   if (ASMJIT_TEST)
+    enable_testing()
     set(ASMJIT_TEST_SRC test/asmjit_test_unit.cpp test/broken.cpp test/broken.h)
 
     asmjit_add_target(asmjit_test_unit EXECUTABLE
@@ -428,6 +429,7 @@ if (NOT ASMJIT_EMBED)
       "${ASMJIT_PRIVATE_CFLAGS_DBG}"
       "${ASMJIT_PRIVATE_CFLAGS_REL}")
     target_compile_definitions(asmjit_test_unit PRIVATE ASMJIT_TEST ASMJIT_STATIC)
+   add_test(asmjit_test_unit asmjit_test_unit)
 
     foreach(_target asmjit_bench_x86
                     asmjit_test_opcode
@@ -440,6 +442,7 @@ if (NOT ASMJIT_EMBED)
         "${ASMJIT_PRIVATE_CFLAGS}"
         "${ASMJIT_PRIVATE_CFLAGS_DBG}"
         "${ASMJIT_PRIVATE_CFLAGS_REL}")
+      add_test(${_target} ${_target})
     endforeach()
   endif()
 endif()
```

**Comment by kobalicek** [comment]
Hi, I looked into it and tried to fix this in https://github.com/asmjit/asmjit/commit/2a088c433a57f75e5f3ad64673b701f3a5cdae7b

Let me know whether that is a sufficient fix.

**Comment by cdluminate** [comment]
Thanks for such an awesome fix!

> Hi, I looked into it and tried to fix this in [2a088c4](https://github.com/asmjit/asmjit/commit/2a088c433a57f75e5f3ad64673b701f3a5cdae7b)
> 
> Let me know whether that is a sufficient fix.
